src/blockchain/rpc.rs

```diff
@@ -0,0 +1,518 @@
+        for wallet_tx in &wallet_txs {
+            let height = match wallet_tx.info.blockhash {
+                Some(blockhash) => Some(
+                    self.0
+                        .as_mut()
+                        .ok_or(Error::OfflineClient)?
+                        .client
+                        .get_block_info(&blockhash)?
+                        .height as u32,
+                ),
+                None => None,
+            };
+            heights.push(height);
+        }
+
+        Ok(txs
+            .iter()
+            .zip(heights.iter())
+            .map(|(tx, height)| (tx.to_owned(), height.to_owned())) // FIXME
+            .collect())
+    }
+}
+
+impl Blockchain for RpcBlockchain {
+    fn offline() -> Self {
+        RpcBlockchain(None)
+    }
+
+    fn is_online(&self) -> bool {
+        self.0.is_some()
+    }
+}
+
+impl OnlineBlockchain for RpcBlockchain {
+    fn get_capabilities(&self) -> HashSet<Capability> {
+        vec![].into_iter().collect()
+    }
+
+    fn setup<D: BatchDatabase + DatabaseUtils, P: Progress>(
+        &mut self,
+        _stop_gap: Option<usize>,
+        _database: &mut D,
+        _progress_update: P,
+    ) -> Result<(), Error> {
+        let config = self.0.as_mut().ok_or(Error::OfflineClient)?;
+
+        // Check we support their node (rust-bitcoincore-rpc supports 0.18.0 and up)
+        let version = config.client.version()?;
+        if version < 180000 {
+            return Err(Error::BitcoinRpcUnsupportedVersion);
+        }
+
+        // Attempt to load watch-only wallet
+        // FIXME: use listwallets once rust-bitcoincore-rpc supports it
+        match config.client.load_wallet(&config.wallet_name) {
+            Ok(_) => info!("Loaded watch-only wallet: \"{}\"", &config.wallet_name),
+            Err(load_wallet_err) => {
+                // Return if watch-only wallet already exists
+                if let RpcError::JsonRpc(jsonrpc::error::Error::Rpc(ref load_wallet_err)) =
+                    load_wallet_err
+                {
+                    if load_wallet_err.message == format!("Wallet file verification failed: Error loading wallet {}. Duplicate -wallet filename specified.", config.wallet_name) {
+                            info!("Watch-only wallet already loaded: \"{}\"", &config.wallet_name);
+                            return Ok(());
+                        }
+                }
+                // Otherwise, create a watch-only wallet
+                match config.client.create_wallet(&config.wallet_name, Some(true)) {
+                    Ok(_) => {
+                        info!("Created watch-only wallet: \"{}\"", &config.wallet_name);
+                    }
+                    Err(error) => {
+                        // FIXME
+                        panic!(
+                            "couldn't create watch-only bitcoin core wallet: {}",
+                            error.to_string()
+                        );
```

Comment:
Don't know what to do here.

- `wallet_name` doesn't exist
- can't create `wallet_name`

Maybe return a `Error::BitcoinRpcCreateWalletFailed` error or something?

Reply:
Yeah, i think it makes sense to have a specific error for this